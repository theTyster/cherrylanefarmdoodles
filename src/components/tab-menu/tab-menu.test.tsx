import { describe, test, expect } from 'vitest';
import TabMenu from './tab-menu';
import React from 'react';

// describe('tabMenu', () => {
//   test('should render correctly', () => {
//     expect(tabMenu({ menuItems: new Map() })).toMatchSnapshot();
//   });
// });
describe('tabMenu', () => {
  test('should render correctly', () => {
    const menuItems = new Map([
      [
        {
          jobHistory: 0,
          0: 'jobHistory',
          initial: 1,
          1: 'initial',
          contributions: 2,
          2: 'contributions',
          hobbies: 3,
          3: 'hobbies',
        },
        [
          {
            id: 0,
            title: 'jobHistory',
            component: <div>jobHistory</div>,
          },
          {
            id: 1,
            title: 'initial',
            component: <div>initial</div>,
          },
          {
            id: 2,
            title: 'contributions',
            component: <div>contributions</div>,
          },
          {
            id: 3,
            title: 'hobbies',
            component: <div>hobbies</div>,
          },
        ],
      ],
    ]);
    expect(<TabMenu menuItems={menuItems} initial='initial' />).toMatchSnapshot();
  });
});


