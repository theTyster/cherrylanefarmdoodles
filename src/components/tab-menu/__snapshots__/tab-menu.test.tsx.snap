// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tabMenu > should render correctly 1`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "initial": "initial",
    "menuItems": Map {
      {
        "0": "jobHistory",
        "1": "initial",
        "2": "contributions",
        "3": "hobbies",
        "contributions": 2,
        "hobbies": 3,
        "initial": 1,
        "jobHistory": 0,
      } => [
        {
          "component": {
            "$$typeof": Symbol(react.transitional.element),
            "_owner": null,
            "_store": {},
            "key": null,
            "props": {
              "children": "jobHistory",
            },
            "type": "div",
          },
          "id": 0,
          "title": "jobHistory",
        },
        {
          "component": {
            "$$typeof": Symbol(react.transitional.element),
            "_owner": null,
            "_store": {},
            "key": null,
            "props": {
              "children": "initial",
            },
            "type": "div",
          },
          "id": 1,
          "title": "initial",
        },
        {
          "component": {
            "$$typeof": Symbol(react.transitional.element),
            "_owner": null,
            "_store": {},
            "key": null,
            "props": {
              "children": "contributions",
            },
            "type": "div",
          },
          "id": 2,
          "title": "contributions",
        },
        {
          "component": {
            "$$typeof": Symbol(react.transitional.element),
            "_owner": null,
            "_store": {},
            "key": null,
            "props": {
              "children": "hobbies",
            },
            "type": "div",
          },
          "id": 3,
          "title": "hobbies",
        },
      ],
    },
  },
  "type": [Function],
}
`;
